osc/util/debquery.py

```diff
@@ -86,6 +72,40 @@ def read(self, all_tags=False, self_provides=True, *extra_tags):
         self.__parse_control(control, all_tags, self_provides, *extra_tags)
         return self
 
+    def __open_tar(self, arfile):
+        control = arfile.get_file(b'control.tar')
+        if control:
+            # XXX: python2.4 relies on a name
+            return tarfile.open(name="control.tar", fileobj=control)
+        return None
+
+    def __open_tar_gz(self, arfile):
+        control = arfile.get_file(b'control.tar.gz')
+        if control:
+            return tarfile.open(name='control.tar.gz', fileobj=control)
+        return None
+
+    def __open_tar_xz(self, arfile):
+        control = arfile.get_file(b'control.tar.xz')
+        if control:
+            if not HAVE_LZMA:
+                raise DebError(self.__path, 'can\'t open control.tar.xz without python-lzma')
+            decompressed = lzma.decompress(control.read())
+            return tarfile.open(name="control.tar.xz",
+                                fileobj=BytesIO(decompressed))
+        return None
```

Comment:
It might be better to reverse the logic (here and in all similar methods):
```
        control = arfile.get_file(b'control.tar.xz')
        if not control:
            return

        if not HAVE_LZMA:
            raise DebError(self.__path, 'can\'t open control.tar.xz without python-lzma')
        decompressed = lzma.decompress(control.read())
        return tarfile.open(name="control.tar.xz", fileobj=BytesIO(decompressed))
```

It saves indentation spaces and makes code a bit more readable.
But I'm rather nit-picking, if you like the existing code, feel free to keep it as is.

Reply:
Looks reasonable. I added 241b34d6 to return early in these functions.